src/scraper/scraper.py

```diff
@@ -0,0 +1,80 @@
+"""
+    Main module interface
+"""
+
+# Local imports
+from scraper.scrapers.base_scraper import BaseScraper
+from .settings import URL_TO_SCRAPER
+from scraper.utils import get_domain_from_url, valid_url
+
+# Python imports
+from typing import List, Type
+
```

Comment:
I know that this is the first step and because of that is ok for me if the following observation is handle in a follow-up task/PR.

Currently we are returning a Dict, but I encourage you to start as soon as possible to work with explicit typed data structures.
More precisely I would like to see a hierarchy of DataClasses representing outputs. And this hierarchy, up to some extent, should be aligned with the current hierarchy you already put in place in this PR.
And, if can be more strict and picky, I would like to see the current hierarchy of classes being defined as Generics which are parametrized by the output DataClasses.
Another important observation is that those DataClasses should be seeing as Inmutable objects. They could have methods performing computations, but they should avoid mutating the object itself. That design should help you to make everything more friendly with pipelining of tasks and future parallelizations.